mailscanner/filter.inc.php

```diff
@@ -232,7 +232,7 @@ public function CreateSQL()
                         $sql .= "AND\n $val[0] $val[1]\n";
                     } else {
                         // Allow !<sql_function>
-                        if ($val[2]{0} === '!') {
+                        if ($val[2]!='' && $val[2]{0} === '!') {
```

Comment:
Agree with the strict comparison.

Not sure I agree with checking  `$val[2]`  with isset.  In this case, `$val[2]` will always be "set" by the post data from the form, so isset will always return true in this case.